### PR TITLE
[12.0][MIG] mass_editing: Add apply domain option in lines

### PR DESCRIPTION
**Pull request by victoralmau**
Add apply domain option in lines to fix some many2one fields with default domain (`company_id` for example)

Please @pedrobaeza  review it.

@Tecnativa TT26786

### CODE REVIEW

File: mass_editing/wizard/mass_editing_wizard.py [hunk 1]
```diff
@@ -76,6 +76,9 @@ def fields_view_get(
             # Field part
             field = line.field_id
             field_info = fields_info[field.name]
+            if not line.apply_domain and 'domain' in field_info:
+                field_info['domain'] = ''
+
```
Comment (pedrobaeza) [inline mass_editing/wizard/mass_editing_wizard.py]: Don't add empty lines inside a method

**Comment by pedrobaeza** [comment]
Please include the infamous widget selection removal. And I think that applying the domain, the fields in that domain should be automatically included in the view.

File: mass_editing/models/mass_editing_line.py [hunk 2]
```diff
@@ -38,6 +38,12 @@ class MassEditingLine(models.Model):
         "'many2many_tags', 'selection'",
     )
 
+    apply_domain = fields.Boolean(
+        default=True,
```
Comment (pedrobaeza) [inline mass_editing/models/mass_editing_line.py]: Please, we don't want to fail by default

```suggestion
        default=False,
```

**Comment by victoralmau** [comment]
> Please include the infamous widget selection removal. And I think that applying the domain, the fields in that domain should be automatically included in the view.

Added widget selection removal. 
IMO isn't necessary to auto-add field included in field domains because not exist apply domain field.

File: mass_editing/models/mass_editing_line.py [hunk 3]
```diff
@@ -38,16 +38,8 @@ class MassEditingLine(models.Model):
...
 
-    @api.depends("field_id")
-    def _compute_widget_option(self):
-        # this function propose selection, depending on the field
-        for line in self.filtered("field_id"):
-            field = line.field_id
-            line.widget_option = False
-            if field.ttype == "many2one":
-                line.widget_option = "selection"
-            elif field.ttype == "many2many":
```
Comment (pedrobaeza) [inline mass_editing/models/mass_editing_line.py:49]: The many2many_tags and the image one are useful, but not the selection one

**Comment by pedrobaeza** [comment]
I still see that fields of the domain must be added to the view if not already in it. Please check it.

File: mass_editing/models/mass_editing_line.py (hunk 3 above)
Comment (pedrobaeza) [reply mass_editing/models/mass_editing_line.py:49]: I think this is bad conceived, as this shouldn't be a computed field, but a stored one with "defaults". And please do it in a separate commit, as they are different changesets from the domain one.

**Comment by victoralmau** [comment]
> I still see that fields of the domain must be added to the view if not already in it. Please check it.

Ok, we auto-add the fields that are necessary for the domain when `apply_domain` is defined.

File: mass_editing/models/mass_editing_line.py (hunk 3 above)
Comment (victoralmau) [reply mass_editing/models/mass_editing_line.py:49]: Try to define `default` in `widget_option` field but need to defined `field_id` previously, therefore it is now defined with an `@api.onchange`

File: mass_editing/readme/ROADMAP.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+- - Auto add fields that is used in domain related if apply_domain=True
```
Comment (pedrobaeza) [inline mass_editing/readme/ROADMAP.rst]:
```suggestion
- Auto add fields that are used in related domains if apply_domain=True
```

File: mass_editing/models/mass_editing.py [hunk 5]
```diff
@@ -14,3 +14,13 @@ class MassEditing(models.Model):
...
+    apply_domain_in_lines = fields.Boolean(
+        string="Apply domain in lines",
+        compute="_compute_apply_domain_in_lines"
+    )
+
+    @api.depends("line_ids")
+    def _compute_apply_domain_in_lines(self):
+        self.apply_domain_in_lines = False
+        if self.line_ids.filtered(lambda x: x.apply_domain):
+            self.apply_domain_in_lines = True
```
Comment (pedrobaeza) [inline mass_editing/models/mass_editing.py]: Not multi-record aware:
```suggestion
        for record in self:
            record.apply_domain_in_lines = any(record.line_ids.mapped("apply_domain"))
```

File: mass_editing/views/mass_editing_view.xml [hunk 6]
```diff
@@ -30,6 +30,12 @@
         <field name="inherit_id" ref="mass_operation_abstract.view_mass_operation_mixin_form"/>
         <field name="mode">primary</field>
         <field name="arch" type="xml">
+            <field name="domain" position="after">
+                <field name="apply_domain_in_lines" invisible="1" />
+                <div colspan="2" class="text-warning" attrs="{'invisible': [('apply_domain_in_lines','=', False)]}">
+                    <b>WARNING</b>: Take into account that adding a field with a domain, and not including the fields of such domain in this operation defintion, will lead to an error when trying to perform it
```
Comment (pedrobaeza) [inline mass_editing/views/mass_editing_view.xml]:
```suggestion
                    <b>WARNING</b>: Take into account that adding a field with a domain, and not including the fields of such domain in this operation definition, will lead to an error when trying to perform it. Make sure you include them.
```

File: mass_editing/wizard/mass_editing_wizard.py [hunk 7]
```diff
@@ -76,6 +76,8 @@ def fields_view_get(
             # Field part
             field = line.field_id
             field_info = fields_info[field.name]
+            if not line.apply_domain and 'domain' in field_info:
+                field_info['domain'] = ''
```
Comment (pedrobaeza) [inline mass_editing/wizard/mass_editing_wizard.py]: Shouldn't this be  `pop field_info['domain']` or `field_info['domain'] = '[]'`?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Review by HaraldPanten — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
/ocabot merge minor

Please fw-port it to v13

**Pull request merged**